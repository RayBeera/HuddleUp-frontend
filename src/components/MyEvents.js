import React, { useState, useEffect } from "react";
import AnchorLink from "react-anchor-link-smooth-scroll-v2";
import axios from "axios";
import EventCard from "./EventCard";
import VotedEventCard from "./VotedEventCard";
import "../styles/my-events.css";

const MyEvents = () => {
  const initialState = {
    votedEvents: [],
    pendingEvents: [],
  };

  const [votedEvents, setVotedEvents] = useState(initialState.votedEvents);
  const [pendingEvents, setPendingEvents] = useState(initialState.pendingEvents);
  const [alert, setAlert] = useState({ message: "" });
  // const [cardRemoved, setCardRemoved] = useState(false);

  const getVotedEvents = (event) => {
    const totalUsers = event.Users.length;
    const suggestionsList = event.Suggestions;
    const totalVotes = suggestionsList.reduce(
      (prev, current) => prev + current.votes, 0
    );
    return totalUsers - totalVotes;
  };

  useEffect(() => {
    axios
      .get("http://localhost:4000/events")
      .then((res) => {
        setVotedEvents(res.data.filter((event) => getVotedEvents(event) === 0));
        setPendingEvents(res.data.filter((event) => getVotedEvents(event) > 0));
      })
      .catch(() => {
        setAlert({
          message: "Server error. Please try again later.",
        });
      });
  }, []);

  return (
    <div className="events">
      <h3> My Events</h3>
      <div className="navigate-events">
        <AnchorLink href='#event-cards-voted'>
          <button>Voting finished</button>
        </AnchorLink>
        <AnchorLink href='#event-cards-pending'>
          <button>Voting in progress</button>
        </AnchorLink>
        <div className="join-create-buttons">
          <button>Join an event</button>
          <button>Create an event</button>
        </div>
      </div>
      <div className="event-cards">
        <div className="event-cards-voted" id="event-cards-voted">
          <h4>Voting Finished</h4>
          {votedEvents && votedEvents.map((votedEvent) => (
            <div className="voted-cards__item" key={votedEvent.id}>
              <VotedEventCard {...votedEvent} />
            </div>
          ))}
        </div>
        <div className="event-cards-pending" id="event-cards-pending">
          <h4>Voting In Progress</h4>
          {pendingEvents && pendingEvents.map((pendingEvent) => (
            <div className="event-cards__item" key={pendingEvent.id}>
              <EventCard {...pendingEvent} />
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default MyEvents;
