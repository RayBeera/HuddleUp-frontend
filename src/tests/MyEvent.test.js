import React from "react";
import renderer from "react-test-renderer";
import { AuthContextProvider } from "../contexts/AuthContext";
import MyEvents from "../components/MyEvents";

describe("MyEvents", () => {
  it("renders correctly", () => {
    const rendered = renderer.create(
      <AuthContextProvider>
        <MyEvents />
      </AuthContextProvider>
    );
    expect(rendered).toMatchSnapshot();
  });
});