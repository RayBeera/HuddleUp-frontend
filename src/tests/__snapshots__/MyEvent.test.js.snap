// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEvents renders correctly 1`] = `
<div
  className="events"
>
  <h3>
     My Events
  </h3>
  <div
    className="navigate-events"
  >
    <a
      href="#event-cards-voted"
      onClick={[Function]}
    >
      <button>
        Voting finished
      </button>
    </a>
    <a
      href="#event-cards-pending"
      onClick={[Function]}
    >
      <button>
        Voting in progress
      </button>
    </a>
    <div
      className="join-create-buttons"
    >
      <button>
        Join an event
      </button>
      <button>
        Create an event
      </button>
    </div>
  </div>
  <div
    className="event-cards"
  >
    <div
      className="event-cards-voted"
      id="event-cards-voted"
    >
      <h4>
        Voting Finished
      </h4>
    </div>
    <div
      className="event-cards-pending"
      id="event-cards-pending"
    >
      <h4>
        Voting In Progress
      </h4>
    </div>
  </div>
</div>
`;
